### PR TITLE
WriteUnPrepared: Add new variable write_batch_flush_threshold

### PR DESCRIPTION
**Pull request by lth**
Instead of reusing `TransactionOptions::max_write_batch_size` for determining when to flush a write batch for write unprepared, add a new variable called `write_batch_flush_threshold` for this use case instead.

Also add `TransactionDBOptions::default_write_batch_flush_threshold` which sets the default value if `TransactionOptions::write_batch_flush_threshold` is unspecified.

### CODE REVIEW

File: include/rocksdb/utilities/transaction_db.h [hunk 1]
```diff
@@ -101,6 +101,11 @@ struct TransactionDBOptions {
   // ordering rather than concurrency control.
   bool skip_concurrency_control = false;
 
+  // This option is only valid for write unprepared. If a write batch exceeds
+  // this threshold, then the transaction will implicitly flush the currently
+  // pending writes into the database. A value of 0 or less means no limit.
+  ssize_t default_write_batch_flush_threshold = 0;
```
Comment (maysamyabandeh) [inline include/rocksdb/utilities/transaction_db.h]: ssize_t is used rarely in the code base. What adv it brings over size_t in this case?

File: utilities/transactions/write_unprepared_txn.cc [hunk 2]
```diff
@@ -71,8 +70,13 @@ WriteUnpreparedTxn::~WriteUnpreparedTxn() {
 
 void WriteUnpreparedTxn::Initialize(const TransactionOptions& txn_options) {
   PessimisticTransaction::Initialize(txn_options);
-  max_write_batch_size_ = txn_options.max_write_batch_size;
-  write_batch_.SetMaxBytes(0);
```
Comment (maysamyabandeh) [inline utilities/transactions/write_unprepared_txn.cc]: Why is this line deleted?

File: utilities/transactions/write_unprepared_txn.h [hunk 3]
```diff
@@ -164,10 +164,10 @@ class WriteUnpreparedTxn : public WritePreparedTxn {
   Status HandleWrite(std::function<Status()> do_write);
 
   // For write unprepared, we check on every writebatch append to see if
-  // max_write_batch_size_ has been exceeded, and then call
+  // write_batch_flush_threshold has been exceeded, and then call
   // FlushWriteBatchToDB if so. This logic is encapsulated in
   // MaybeFlushWriteBatchToDB.
-  size_t max_write_batch_size_;
+  ssize_t write_batch_flush_threshold_;
```
Comment (maysamyabandeh) [inline utilities/transactions/write_unprepared_txn.h]: Why does this have to be ssize_t?

File: utilities/transactions/write_unprepared_txn.h [hunk 4]
```diff
@@ -164,10 +164,10 @@ class WriteUnpreparedTxn : public WritePreparedTxn {
   Status HandleWrite(std::function<Status()> do_write);
 
   // For write unprepared, we check on every writebatch append to see if
-  // max_write_batch_size_ has been exceeded, and then call
+  // write_batch_flush_threshold has been exceeded, and then call
```
Comment (maysamyabandeh) [inline utilities/transactions/write_unprepared_txn.h]: Do we need underscore after write_batch_flush_threshold?

File: include/rocksdb/utilities/transaction_db.h (hunk 1 above)
Comment (lth) [reply include/rocksdb/utilities/transaction_db.h]: This is because `TransactionOptions::write_batch_flush_threshold` is signed. A value of -1 there means "unspecified".

If `default_write_batch_flush_threshold` were unsigned, then we would have to be more careful when assigning `default_write_batch_flush_threshold` onto `default_write_batch_flush_threshold`, because the max of size_t is larger than ssize_t.

I guess I could add validation logic in `WriteUnpreparedTxnDB::Initialize` to check that `default_write_batch_flush_threshold` is less than max of ssize_t.

File: utilities/transactions/write_unprepared_txn.cc (hunk 2 above)
Comment (lth) [reply utilities/transactions/write_unprepared_txn.cc]: We're setting the max write batch size to infinite here. However, the goal of this change is that we want write_batch_ to have whatever limit is specified in `txn_options.max_write_batch_size`, to have consistent behaviour with other write policies. This happens in `PessimisticTransaction::Initialize`.

File: utilities/transactions/write_unprepared_txn.h (hunk 3 above)
Comment (lth) [reply utilities/transactions/write_unprepared_txn.h]: Same reason as above.
